contracts/IPool.sol

```diff
@@ -7,13 +7,24 @@ pragma solidity ^0.8.0;
  */
 abstract contract IPool {
 
-    function deposit(uint256 amount) external virtual;
+    function depositUSDC(
```

Comment:
can we make the functionality more generic? why limit it to just USDC. The pool itself can choose to use USDC. Also just out of curiosity why use abstract instead of an interface

Reply:
> can we make the functionality more generic?

Yes, will try to do so

> Also just out of curiosity why use abstract instead of an interface

Abstracts are more flexible on their limitations. For instance, an interface in Solidity can't have any implemented functions or enums, they can't inherit from other contracts, they can't have state variables, etc.